lib/corefoundation/exceptions.rb

```diff
@@ -0,0 +1,16 @@
+module CF
+  module Exceptions
+    # Raised when the preference value couldn't be found for a domain.
+    class PreferenceDoesNotExist < StandardError
+      def initialize(key, domain, hostname)
```

Comment:
`hostname` should be optional

Reply:
Yep. Been busy for a while so couldn't get back to the code. Updating this now.